core/src/main/java/org/apache/iceberg/util/ParallelIterable.java

```diff
@@ -53,7 +57,8 @@ public CloseableIterator<T> iterator() {
     private final Iterator<Runnable> tasks;
     private final ExecutorService workerPool;
     private final Future<?>[] taskFutures;
-    private final ConcurrentLinkedQueue<T> queue = new ConcurrentLinkedQueue<>();
+    private final LinkedBlockingQueue<T> queue =
+        new LinkedBlockingQueue<>(MANIFEST_READER_QUEUE_SIZE);
```

Comment:
This config seems to affect all the instances.

Reply:
iceberg does not seem to provide a more effective configuration management method other than environment variables. Can we turn this into a table-level configuration?

Reply:
@Heltman, we typically avoid system or environment config. A more appropriate place for this is in the engine's config. Then it can pass that configuration down. For example, Flink passes threadpools into the scan API rather than using the common worker pool.

Reply:
How I pass a threadpools to iceberg from trino. Trino only use `plan()` to get FileScanTask